src/v/bytes/details/io_allocation_size.h

```diff
@@ -29,28 +29,40 @@ class io_allocation_size {
     static constexpr size_t ss_max_small_allocation = 16384;
 
 public:
-    // >>> x=512
-    // >>> while x < int((1024*128)):
-    // ...     print(x)
-    // ...     x=int(((x*3)+1)/2)
-    // ...     x=int(min(1024*128,x))
-    // print(1024*128)
+    // This script computes the table immediately below, which are "ideal"
+    // allocation sizes: rounded up to the next true size supported by
+    // the seastar allocator. At <= 16K we apply the small pool indexing
+    // logic, and above we use powers of 2 since we are in the buddy allocator
+    // which only supports power-of-two sizes.
+    //
+    // We scale the target size by 1.47, i.e., 1.5 tweaked slightly to ensure
+    // we hit 16K at the small<->big pool boundary.
+    //
+    // def lg2(size: int):
+    //     return size.bit_length() - 1
+    // def p(v: object):
+    //     print(f"{v},")
+    // s = 512
+    // fb = 2 # idx_frac_bits
+    // while s <= 2**14:
+    //     idx = ((lg2(s) << fb) - ((1 << fb) - 1)) + ((s - 1) >> (lg2(s) - fb))
```

Comment:
nit: add link to https://github.com/scylladb/seastar/blob/f840b860432e7e716e3cfc004690897b50dc122c/src/core/memory.cc#L495-L499 in comment for reference?

Reply:
Good idea, added this comment:

https://github.com/redpanda-data/redpanda/pull/24475/files#diff-c87b358938b6d0617da0eb7cde237236bc36df34121586361159a56d3c38dbeaR47-R53

Reply:
Thanks!